src/policy/concrete.rs

```diff
@@ -41,6 +42,14 @@ use crate::miniscript::limits::{LOCKTIME_THRESHOLD, SEQUENCE_LOCKTIME_TYPE_FLAG}
 use crate::miniscript::types::extra_props::TimelockInfo;
 use crate::{errstr, Error, ForEach, ForEachKey, MiniscriptKey};
 
+/// [`TapTree`] -> ([`Policy`], satisfaction cost) cache
+#[cfg(feature = "compiler")]
+type PolicyTapCache<Pk> = BTreeMap<TapTree<Pk>, (Policy<Pk>, f64)>;
+
+/// [`Miniscript`] -> leaf probability in policy cache
+#[cfg(feature = "compiler")]
+type MsTapCache<Pk> = BTreeMap<TapTree<Pk>, f64>;
```

Comment:
What happens when there is a collision? Making sure that we are dealing with those cases in a sane manner.

Reply:
Sure I'll update this! This was done considering that there were no duplicate keys in the initial policy.

Reply:
There are no duplicates, but still, there can be collisions in the compilation.

Reply:
Right, that makes sense!